### PR TITLE
Added AVL Tree implementation

### PR DESCRIPTION
**Pull request by imsushant12**
## Related Issue 

Issue Number : #78 

Closes: #78 

### Describe the changes you've made

I have added the implementation of the AVL Tree along with sample output and Time Complexity in C Language.

## Type of change

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

I have verified it with sample input and output.

### CODE REVIEW

**Comment by himabindu-run** [comment]
@imsushant12 There is a bug, while we give a string as input, the code is going into an Infinite loop!
Example: I gave "wercg" as an element to Insert in the tree. This is the case, can please fix the bug and let us know?

image

@tireless-22 Take a look please

**Comment by imsushant12** [comment]
> @imsushant12 There is a bug, while we give a string as input, the code is going into an Infinite loop! Example: I gave "wercg" as an element to Insert in the tree. This is the case, can please fix the bug and let us know?
> 
> image
> 
> @tireless-22 Take a look please

It is supoosed to work on integers only as I have taken int.

**Comment by himabindu-run** [comment]
@imsushant12 Yes, I understood that but can you do something to alert/indicate the user to not give strings or handle that condition?

File: Tree/AVL_Trees/AVL_Tree.c [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+
+
+
+int main()
+{
+    struct node *root = NULL;
+    int data,choice;
+    while(1)
+    {
+        printf("\n1. Insertion");
```
Comment (himabindu-run) [inline Tree/AVL_Trees/AVL_Tree.c:280]: You can do something here, or the function can be modified

Comment (imsushant12) [reply Tree/AVL_Trees/AVL_Tree.c:280]: I actually added a message to input integer only. Will it be okay?

**Comment by deephunt3r** [comment]
@himabindu-run please have a look

**Review by himabindu-run — COMMENTED** [review]
LGTM 🚀

**Pull request merged**